src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs

```diff
@@ -0,0 +1,272 @@
+        {
+            { WellKnownTags.Public, CompletionItemKind.Keyword },
+            { WellKnownTags.Protected, CompletionItemKind.Keyword },
+            { WellKnownTags.Private, CompletionItemKind.Keyword },
+            { WellKnownTags.Internal, CompletionItemKind.Keyword },
+            { WellKnownTags.File, CompletionItemKind.File },
+            { WellKnownTags.Project, CompletionItemKind.File },
+            { WellKnownTags.Folder, CompletionItemKind.Folder },
+            { WellKnownTags.Assembly, CompletionItemKind.File },
+            { WellKnownTags.Class, CompletionItemKind.Class },
+            { WellKnownTags.Constant, CompletionItemKind.Constant },
+            { WellKnownTags.Delegate, CompletionItemKind.Function },
+            { WellKnownTags.Enum, CompletionItemKind.Enum },
+            { WellKnownTags.EnumMember, CompletionItemKind.EnumMember },
+            { WellKnownTags.Event, CompletionItemKind.Event },
+            { WellKnownTags.ExtensionMethod, CompletionItemKind.Method },
+            { WellKnownTags.Field, CompletionItemKind.Field },
+            { WellKnownTags.Interface, CompletionItemKind.Interface },
+            { WellKnownTags.Intrinsic, CompletionItemKind.Text },
+            { WellKnownTags.Keyword, CompletionItemKind.Keyword },
+            { WellKnownTags.Label, CompletionItemKind.Text },
+            { WellKnownTags.Local, CompletionItemKind.Variable },
+            { WellKnownTags.Namespace, CompletionItemKind.Text },
+            { WellKnownTags.Method, CompletionItemKind.Method },
+            { WellKnownTags.Module, CompletionItemKind.Module },
+            { WellKnownTags.Operator, CompletionItemKind.Operator },
+            { WellKnownTags.Parameter, CompletionItemKind.Value },
+            { WellKnownTags.Property, CompletionItemKind.Property },
+            { WellKnownTags.RangeVariable, CompletionItemKind.Variable },
+            { WellKnownTags.Reference, CompletionItemKind.Reference },
+            { WellKnownTags.Structure, CompletionItemKind.Struct },
+            { WellKnownTags.TypeParameter, CompletionItemKind.TypeParameter },
+            { WellKnownTags.Snippet, CompletionItemKind.Snippet },
+            { WellKnownTags.Error, CompletionItemKind.Text },
+            { WellKnownTags.Warning, CompletionItemKind.Text },
+        };
+
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly FormattingOptions _formattingOptions;
+        private readonly ILogger _logger;
+
+        private (CSharpCompletionList Completions, string FileName, int Position)? _lastCompletion = null;
+
+        [ImportingConstructor]
+        public CompletionService(OmniSharpWorkspace workspace, FormattingOptions formattingOptions, ILoggerFactory loggerFactory)
+        {
+            _workspace = workspace;
+            _formattingOptions = formattingOptions;
```

Comment:
I now realized that the usage of `FormattingOptions` isn't optimal here because it wouldn't observe editorconfig, but this is something that we broke when we moved to Roslyns' editorcconfig so unrelated to this PR

Reply:
Should I be using something else? It's only used for formatting the markdown popups, so observing editorconfig isn't particularly important for this bit.

Reply:
ah I now see only newline is used, then it doesn't matter anyway.
it used to be so that `FormattingOptions` was also populated by editorconfig settings, but not anymore. it would be more important if we ran the code through the C# formatter/code clean up before putting it in markdown but I don't think even VS does that